### ISSUE TITLE
GLboolean scoped enum?

### ISSUE THREAD

**Issue by flv0** [cginternals/glbinding#175]
Hi, I'm dealing with code which compares against returned GLboolean values, such as
```
assert (glIsRenderbuffer (depthOrStencilBuffer));
```
This will not compile since although being an unsigned char internally, scoped enums won't implicitly convert to int/bool. I see the use of using scoped enums for th GLenum and GLextension types and like the type safety glbinding gives very much, however I wonder why you chose to do it for GLboolean as well, where I think people want to use GLbooleans in conditional expressions and be able to set them to true/false. Maybe an alternative could be to define operator! for the scoped enum type?

**Comment by cgcostume** [comment]
We have another idea here: a class is used instead of an enum class. This class then provides two singletons for ```GL_FALSE``` and ```GL_TRUE``` and is enabled to resolve to boolean via ```operator bool``` (since an enum class cannot be extended by such operator). So every use of ```GL_FALSE``` and  ```GL_TRUE``` will be typesafe and resolve to the two, global single instances of the glBoolean class.

Other solutions *we dismissed* are (1) overloading the ```operator!``` and (2) using a wrapper to provide the implicit conversion (e.g., using the ```operator&``` http://stackoverflow.com/a/12063354)

**Comment by flv0** [comment]
Interesting solution! Looking at the diff it seems it should transparently convert to/from bool now and permit all the logical operations, nice. If there are any probloms with this I'll report back. Just wondering because I don't understand it fully, are the other integral type conversions strictly necessary, or would an operator bool suffice (which would then be converted implicitly to char/int/long)? Thanks!

**Comment by scheibel** [comment]
As you have noticed, we added several cast operators to the ```GLboolean``` class. They are all explicit in means of you have to add a ```static_cast``` to obtain a cast to the integral types. The main type that is relevant is ```bool``` but without the other casts you would have to write ```static_cast<int>(static_cast<bool>(GL_TRUE))``` if you want to access the integer value of ```GL_TRUE```. One really nice situation are control flow structures as ```if``` and ```while``` as they implicitly perform the explicit ```bool``` cast so that using ```GLboolean``` integrates naturally:
```cpp
GLboolean isShader = glIsShader(shader);
if (isShader)
{
    //
}
```

**Comment by cgcostume** [comment]
added minor fix for msvc W4 in 8ebf9498258337b98ed5f6fc957d0d7791a8d7be
